### PR TITLE
Add support for capabilities substitutions

### PR DESCRIPTION
**Pull request by melix**
This is a follow-up to #13286 which does the same, but for capabilities instead of attributes.

### CODE REVIEW

**Comment by melix** [comment]
@bot-gradle test RFM

**Comment by melix** [comment]
@bot-gradle test RFM

**Comment by melix** [comment]
@bot-gradle test RFM

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/VariantSelectionDetails.java [hunk 1]
```diff
@@ -46,9 +46,16 @@
...
-     * in the configuration action.
+     * via the configuration action.
      * @param configurationAction the configurationa action
      */
     void attributes(Action<? super AttributeContainer> configurationAction);
 
+    /**
+     * Replaces the provided selector capabilities with the capabilities configured
+     * via the configuration action.
+     * @param configurationAction the configuraiton action
```
Comment (ljacomet) [inline subprojects/core-api/src/main/java/org/gradle/api/artifacts/VariantSelectionDetails.java:57]: typo in configuration

Comment (melix) [reply subprojects/core-api/src/main/java/org/gradle/api/artifacts/VariantSelectionDetails.java]: Fat fingers :)

**Pull request merged**